### PR TITLE
[Do Not Merge] Adding Private Endpoints (formerly Interface Endpoints)

### PR DESCRIPTION
**Pull request by rupalivohra**
### Reviewers Please Note
I'm creating this PR for ARM reviewers to have extra clarity on the changes to the resources.
Please do not review this PR - it can't be checked in until the network-march-release branch is ready.

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/privateEndpoint.json [hunk 1]
```diff
@@ -50,11 +50,11 @@
             "description": "The name of the resource group."
           },
           {
-            "name": "interfaceEndpointName",
+            "name": "privateEndpointName",
             "in": "path",
             "required": true,
             "type": "string",
-            "description": "The name of the interface endpoint."
+            "description": "The name of the private endpoint."
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/privateEndpoint.json]: Please add the "default" response to model errors, i.e. :
``
          "default": {
            "description": "Error",
            "schema": {
              "$ref": "./network.json#/definitions/Error"
            }
          }
``

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/privateEndpoint.json [hunk 2]
```diff
@@ -116,22 +116,22 @@
         ],
         "responses": {
           "200": {
-            "description": "Request successful. The operation returns the resulting InterfaceEndpoint resource.",
+            "description": "Request successful. The operation returns the resulting PrivateEndpoint resource.",
             "schema": {
-              "$ref": "#/definitions/InterfaceEndpoint"
+              "$ref": "#/definitions/PrivateEndpoint"
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/privateEndpoint.json]: Please add the "default" response to model errors, i.e. :
``
          "default": {
            "description": "Error",
            "schema": {
              "$ref": "./network.json#/definitions/Error"
            }
          }
``

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/privateEndpoint.json [hunk 3]
```diff
@@ -165,31 +165,31 @@
         ],
         "responses": {
           "200": {
-            "description": "Update successful. The operation returns the resulting InterfaceEndpoint resource.",
+            "description": "Update successful. The operation returns the resulting PrivateEndpoint resource.",
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/privateEndpoint.json]: See above regarding "default" response

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/privateEndpoint.json [hunk 4]
```diff
@@ -271,17 +271,55 @@
...
+            {
+              "$ref": "./network.json#/definitions/SubResource"
+            }
+          ],
+          "description": "A grouping of information about the connection to the remote resource."
         },
-        "endpointService": {
-          "$ref": "#/definitions/EndpointService",
-          "description": "A reference to the service being brought into the virtual network."
+        "manualPrivateLinkServiceConnection": {
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/privateEndpoint.json]: Is this used? Is it intended to be identical to `privateLinkServiceConnection`

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see the few comments. Thanks

**Comment by MikhailTryakhov** [comment]
closed due to milestone was missed. please retarget to newer branch when it's ready

**Pull request closed without merging**